activerecord/lib/active_record/database_configurations/hash_config.rb

```diff
@@ -158,6 +158,12 @@ def schema_dump(format = ActiveRecord.schema_format)
         end
       end
 
+      def schema_format(check_env: false)
+        return ENV["SCHEMA_FORMAT"] if check_env && ENV["SCHEMA_FORMAT"].present?
```

Comment:
Reference to the `SCHEMA_FORMAT` environment variable is limited to the database tasks today. It feels like it's an unnecessary complexity here. Do we really need to move it here?

Reply:
Fair enough, I'll keep them over there.